crates/matrix-sdk/src/send_queue/mod.rs

```diff
@@ -1589,22 +1599,28 @@ impl QueueStorage {
                 }
             }
 
-            DependentQueuedRequestKind::UploadFileWithThumbnail {
+            DependentQueuedRequestKind::UploadFileOrThumbnail {
                 content_type,
                 cache_key,
                 related_to,
+                #[cfg(feature = "unstable-msc4274")]
+                depends_on_thumbnail,
             } => {
                 let Some(parent_key) = parent_key else {
                     // Not finished yet, we should retry later => false.
                     return Ok(false);
                 };
-                self.handle_dependent_file_upload_with_thumbnail(
+                self.handle_dependent_file_or_thumbnail_upload(
                     client,
                     dependent_request.own_transaction_id.into(),
                     parent_key,
                     content_type,
                     cache_key,
                     related_to,
+                    #[cfg(feature = "unstable-msc4274")]
+                    depends_on_thumbnail,
+                    #[cfg(not(feature = "unstable-msc4274"))]
+                    true,
```

Comment:
To make it easier to read, can you define a local variable `depends_on_thumbnail` and define it accordingly? (maybe with `cfg_if`)